debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerDefaultValueConverter.java

```diff
@@ -0,0 +1,190 @@
+        result.put("smallmoney", v -> new BigDecimal(v.substring(2, v.length() - 2))); // Sample value: ((214748.3647))
+
+        // Approximate numerics
+        result.put("float", v -> Double.parseDouble(v.substring(2, v.length() - 2))); // Sample value: ((1.2345000000000000e+003))
+        result.put("real", v -> Float.parseFloat(v.substring(2, v.length() - 2))); // Sample value: ((1.2345000000000000e+003))
+
+        // Date and time
+        result.put("date", v -> { // Sample value: ('2019-02-03')
+            String rawValue = v.substring(2, v.length() - 2);
+            return querySingleValue("SELECT PARSE(? AS date)", st -> st.setString(1, rawValue), rs -> rs.getDate(1));
+        });
+        result.put("datetime", v -> { // Sample value: ('2019-01-01 00:00:00.000')
+            String rawValue = v.substring(2, v.length() - 2);
+            return querySingleValue("SELECT PARSE(? AS datetime)", st -> st.setString(1, rawValue), rs -> rs.getTimestamp(1));
+        });
+        result.put("datetime2", v -> { // Sample value: ('2019-01-01 00:00:00.1234567')
+            String rawValue = v.substring(2, v.length() - 2);
+            return querySingleValue("SELECT PARSE(? AS datetime2)", st -> st.setString(1, rawValue), rs -> rs.getTimestamp(1));
+        });
+        result.put("datetimeoffset", v -> { // Sample value: ('2019-01-01 00:00:00.1234567+02:00')
+            String rawValue = v.substring(2, v.length() - 2);
+            return querySingleValue("SELECT PARSE(? AS datetimeoffset)", st -> st.setString(1, rawValue), rs -> (DateTimeOffset) rs.getObject(1));
+        });
+        result.put("smalldatetime", v -> { // Sample value: ('2019-01-01 00:00:00')
+            String rawValue = v.substring(2, v.length() - 2);
+            return querySingleValue("SELECT PARSE(? AS smalldatetime)", st -> st.setString(1, rawValue), rs -> rs.getTimestamp(1));
+        });
+        result.put("time", v -> { // Sample value: ('2019-01-01 00:00:00')
+            String rawValue = v.substring(2, v.length() - 2);
+            return querySingleValue("SELECT PARSE(? AS time)", st -> st.setString(1, rawValue), rs -> rs.getTime(1));
+        });
+
+        // Character strings
+        result.put("char", v -> v.substring(2, v.length() - 2)); // Sample value: ('aaa')
+        result.put("text", v -> v.substring(2, v.length() - 2)); // Sample value: ('aaa')
+        result.put("varchar", v -> v.substring(2, v.length() - 2)); // Sample value: ('aaa')
+
+        // Unicode character strings
+        result.put("nchar", v -> v.substring(2, v.length() - 2)); // Sample value: ('aaa')
+        result.put("ntext", v -> v.substring(2, v.length() - 2)); // Sample value: ('aaa')
+        result.put("nvarchar", v -> v.substring(2, v.length() - 2)); // Sample value: ('aaa')
+
+        // Binary strings
+        result.put("binary", v -> HexConverter.convertFromHex(v.substring(3, v.length() - 1))); // Sample value: (0x0102030405)
```

Comment:
It seems we don't have any tests for these types (unrelated to the issue at hand). Could you double-check and log an issue if you confirm?

Reply:
https://issues.redhat.com/browse/DBZ-2174